### PR TITLE
Fixes # 24 : Setup automatic deployment for public AI Lab site

### LINKED ISSUE #15: Triggering website deployment with Github Actions
Prerequisite for having an alpha website: https://alpha.canada.ca/fr/instructions.html

Info: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

### LINKED ISSUE #24: Setup automatic deployment for public AI Lab site
- Create gh-pages branch
- Create workflow for build
- Create workflow for deployment

### PR DESCRIPTION
**Pull request by vivalareda**
Added the yml file for automatic deployment
Fixes #24
also fixes #15

### CODE REVIEW

File: .github/workflows/gh-pages-deployment.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: ./ai-cfia-ia-acia/build
+          publish_branch: deployment
+
+          user_name: github-actions[bot]
+          user_email: 41898282+github-actions[bot]@users.noreply.github.com
```
Comment (rngadam) [inline .github/workflows/gh-pages-deployment.yml]: can you explain the choice of email?

File: .github/workflows/gh-pages-deployment.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: yarn
+          cache-dependency-path: 'ai-cfia-ia-acia'
+
```
Comment (rngadam) [inline .github/workflows/gh-pages-deployment.yml]: extra whitelines

File: .github/workflows/gh-pages-deployment.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    name: Deploy to GitHub Pages
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 18
+          cache: yarn
+          cache-dependency-path: 'ai-cfia-ia-acia'
```
Comment (rngadam) [inline .github/workflows/gh-pages-deployment.yml]: repeated constant can be pulled up to env: directive even in a non-reusable workflow.

File: .github/workflows/gh-pages-deployment.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+name: Deploy to GitHub Pages
+
+on:
+  push:
+    branches:
+      - redat97/issue24
+
+permissions:
+  contents: write
```
Comment (rngadam) [inline .github/workflows/gh-pages-deployment.yml:9]: could deployments: true work here?

https://docs.github.com/en/actions/using-jobs/assigning-permissions-to-jobs

**Review by rngadam — APPROVED** [review]

**Review by rngadam — CHANGES_REQUESTED** [review]
oops approved accidentally

File: .github/workflows/gh-pages-deployment.yml (hunk 1 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml]: it's default github action bot email
https://github.com/marketplace/actions/bot-details 
and is used in the example in docusaurus documentation
https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

File: .github/workflows/gh-pages-deployment.yml (hunk 4 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml:9]: it's what they use in the example of docusaurus 
https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

I don't know how I can try it out with website down but I think the permission you are asking for isn't least privilege I but I could be wrong, if I understood correctly, the content privilege only pushs changes to the repository it doesn't actually deploy

File: .github/workflows/gh-pages-deployment.yml (hunk 2 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml]: done

File: .github/workflows/gh-pages-deployment.yml (hunk 3 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml]: done

**Comment by vivalareda** [comment]
I changed the branch on push main also

File: .github/workflows/gh-pages-deployment.yml (hunk 1 above)
Comment (rngadam) [reply .github/workflows/gh-pages-deployment.yml]: can you put this information in a comment in the file? usually if I ask for clarification it is meant for you to clarify the code.

**Review by rngadam — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages-deployment.yml (hunk 1 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml]: ah yeah make sense mb

Comment (rngadam) [reply .github/workflows/gh-pages-deployment.yml]: don't just put a end-of-line comment; create a real comment with the links you posted here

Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml]: done

File: .github/workflows/gh-pages-deployment.yml [hunk 5]
```diff
@@ -45,4 +45,8 @@ jobs:
           publish_branch: deployment
 
           user_name: github-actions[bot]
-          user_email: 41898282+github-actions[bot]@users.noreply.github.com # default github-actions email
\ No newline at end of file
+          user_email: 41898282+github-actions[bot]@users.noreply.github.com
+          # The email used below is the default GitHub Action bot email.
```
Comment (rngadam) [inline .github/workflows/gh-pages-deployment.yml:48]: inconsistent indent in the block below. put the comment block above what you're commenting

**Review by rngadam — CHANGES_REQUESTED** [review]

File: .github/workflows/gh-pages-deployment.yml (hunk 5 above)
Comment (vivalareda) [reply .github/workflows/gh-pages-deployment.yml:48]: sorry about that, I indented it and changed placement, I am not sure if it's better to leave a blank like between user_email and start of comment for readability. I opted for not but if you think it's better to do so lmk

**Comment by rngadam** [comment]
@redat97 can you squash your PR commit history into a single commit? there's a lot of commits for the purpose of testing.

**Review by rngadam — CHANGES_REQUESTED** [review]
rework commit to mention issue and get rid of extraneous text that lack meaning

**Review by rngadam — APPROVED** [review]

**Pull request merged**